src/AggregateFunctions/AggregateFunctionSumCount.h

```diff
@@ -0,0 +1,277 @@
+#pragma once
+
+#include <experimental/type_traits>
+#include <type_traits>
+
+#include <IO/WriteHelpers.h>
+#include <IO/ReadHelpers.h>
+
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypesDecimal.h>
+#include <Columns/ColumnVector.h>
+
+#include <AggregateFunctions/IAggregateFunction.h>
+
+
+namespace DB
+{
+
+/// Uses addOverflow method (if available) to avoid UB for sumWithOverflow()
+///
+/// Since NO_SANITIZE_UNDEFINED works only for the function itself, without
+/// callers, and in case of non-POD type (i.e. Decimal) you have overwritten
+/// operator+=(), which will have UB.
+template <typename T>
+struct AggregateFunctionSumCountCountAddOverflowImpl
+{
+    static void NO_SANITIZE_UNDEFINED ALWAYS_INLINE add(T & lhs, const T & rhs)
+    {
+        lhs += rhs;
+    }
+};
+template <typename DecimalNativeType>
+struct AggregateFunctionSumCountCountAddOverflowImpl<Decimal<DecimalNativeType>>
+{
+    static void NO_SANITIZE_UNDEFINED ALWAYS_INLINE add(Decimal<DecimalNativeType> & lhs, const Decimal<DecimalNativeType> & rhs)
+    {
+        lhs.addOverflow(rhs);
+    }
+};
+
+template <typename T>
+struct AggregateFunctionSumCountData
```

Comment:
You're copying the entire code of AggregateFunctionSumData here, right? Did you consider using `AggregateFunctionAvg` instead? It already has the data you need, and you'll only have to overload `getReturnType` and `insertResultInto` so that they return a tuple.

Reply:
ok